sopel/modules/clock.py

```diff
@@ -55,72 +57,89 @@ def setup(bot):
     bot.config.define_section('clock', TimeSection)
 
 
-@commands('t', 'time')
-@example('.t America/New_York')
+@module.commands('t', 'time')
+@module.example('.t America/New_York')
+@module.example('.t Exirel')
+@module.example('.t #sopel')
 def f_time(bot, trigger):
-    """Returns the current time."""
+    """Returns the current time.
+
+    The command takes an optional parameter: it can be either a timezone, a
+    nick, or a channel.
+    """
     if trigger.group(2):
-        zone = get_timezone(bot.db, bot.config, trigger.group(2).strip(), None, None)
-        if not zone:
-            bot.say('Could not find timezone %s.' % trigger.group(2).strip())
-            return
+        argument = trigger.group(2).strip()
+        channel_or_nick = tools.Identifier(argument)
+
+        if channel_or_nick in bot.users:
+            zone = get_nick_timezone(bot.db, channel_or_nick)
+            if zone is None:
+                bot.say('Could not find timezone for nick %s.' % argument)
+                return
+        elif channel_or_nick in bot.channels:
+            zone = get_channel_timezone(bot.db, channel_or_nick)
+            if zone is None:
+                bot.say('Could not find timezone for channel %s.' % argument)
+                return
+        else:
+            try:
+                zone = validate_timezone(argument)
+            except ValueError:
+                bot.say('Could not find timezone %s.' % argument)
+                return
     else:
         zone = get_timezone(bot.db, bot.config, None, trigger.nick,
                             trigger.sender)
+
     time = format_time(bot.db, bot.config, zone, trigger.nick, trigger.sender)
     bot.say(time)
 
 
-@commands('settz', 'settimezone')
-@example('.settz America/New_York')
+@module.commands('settz', 'settimezone')
+@module.example('.settz America/New_York')
 def update_user(bot, trigger):
     """
     Set your preferred time zone. Most timezones will work, but it's best to
     use one from <https://sopel.chat/tz>.
     """
-    if not pytz:
-        bot.reply("Sorry, I don't have timezone support installed.")
+    tz = trigger.group(2)
+    if not tz:
+        bot.reply("What timezone do you want to set? Try one from "
+                  "https://sopel.chat/tz")
+        return
+
+    if tz not in pytz.all_timezones:
```

Comment:
How about taking advantage of `tools.time.validate_timezone()` here? That will make lazy commands like `.settz america/chicago` Just Work™, reducing effort for the user. (Totally fine if you want to make that a separate PR, but this one is already touching quite a bit more than just the behavior of `.t`. Honestly, we should probably give this PR a broader title. I'll need to remember API additions for the changelog entry, as well as the bugfix/tweak.)

Reply:
Yeah I went for the change you suggested. It's easier to do that in one go, in the end.